src/main/java/org/asamk/signal/commands/AttachmentCommand.java

```diff
@@ -0,0 +1,65 @@
+package org.asamk.signal.commands;
+
+import net.sourceforge.argparse4j.inf.Namespace;
+import net.sourceforge.argparse4j.inf.Subparser;
+
+import org.asamk.signal.commands.exceptions.CommandException;
+import org.asamk.signal.commands.exceptions.UnexpectedErrorException;
+import org.asamk.signal.json.JsonAttachmentData;
+import org.asamk.signal.manager.AttachmentPointer;
+import org.asamk.signal.manager.Manager;
+import org.asamk.signal.output.JsonWriter;
+import org.asamk.signal.output.OutputWriter;
+import org.asamk.signal.output.PlainTextWriter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Base64;
+
+public class AttachmentCommand implements JsonRpcLocalCommand {
+
+    @Override
+    public String getName() {
+        return "attachment";
+    }
+
+    @Override
+    public void attachToSubparser(final Subparser subparser) {
+        subparser.addArgument("--file-id")
+                .help("The ID of the attachment file.");
+        subparser.addArgument("--file-name")
+                .nargs("?")
+                .help("The name of the file.");
+        subparser.addArgument("--content-type")
+                .nargs("?")
+                .help("The content type of the file.");
```

Comment:
With my suggestion from above, the file-name and content-type won't be required here anymore.
Instead it would be good to add the recipient number or groupId as required parameter. Currently the attachment folder contains the attachments directly, but that might change in the future, to have them in subfolders by recipient/group.

```suggestion
        subparser.addArgument("id").required(true).help("The ID of the attachment file.");
        var mut = subparser.addMutuallyExclusiveGroup().required(true);
        mut.addArgument("--recipient").help("Sender of the attachment");
        mut.addArgument("-g", "--group-id").help("Group in which the attachment was received");
```

Reply:
Changed that.